### PR TITLE
Notification: fix addObserver(forName:object:queue:using:) signature

### PR DESCRIPTION
**Pull request by ianpartridge**
This is a pickup of https://github.com/apple/swift-corelibs-foundation/pull/1060 which has gone stale.

Correct the API signature of 
https://developer.apple.com/documentation/foundation/notificationcenter/1411723-addobserver

### CODE REVIEW

File: Foundation/NSNotification.swift [hunk 1]
```diff
@@ -187,8 +187,13 @@ open class NotificationCenter: NSObject {
             self._observers = _observers.filterOutObserver(observer, name: aName, object: object)
         })
     }
-    
+
+    @available(swift, obsoleted: 4.0)
```
Comment (alblue) [inline Foundation/NSNotification.swift]: We can get client-side fixes if we annotate it with a 'renamed' attribute

`@available(*,unavailable,renamed:"addObserver(forName:object:queue:using:)")`

Comment (ianpartridge) [inline Foundation/NSNotification.swift]: Nice. So two `@available` attributes on the same method?

Comment (alblue) [inline Foundation/NSNotification.swift]: Although you can have two `@available` attributes, you can also combine them for different platforms:

https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/Attributes.html

So `@available(*,obsoleted:4.0,renamed:"addObserver(forName:object:queue:using:)")` will work in this case.

**Comment by ianpartridge** [comment]
@swift-ci please test

**Comment by alblue** [comment]
@phausler Is this the right way to indicate a renamed method, or should there be some kind of alternate way of representing changes due to the way methods are imported?

**Comment by phausler** [comment]
This is probably the closest we have to a solution that does not break third party code

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Pull request merged**